### ISSUE TITLE
feature: Add a stoke on a specific border

### ISSUE THREAD

**Issue by nina-alin** [uxbox/uxbox#3005]
### Is your feature request related to a problem? Please describe.

Sometimes, when I start to design a component, I want to apply stoke to a specific border of my element.

For example, if I have a tab, I want to apply stoke only at the bottom of the element. At the moment, you can only apply stoke on the full element. 

### Describe the solution you'd like.

Figma has a select element with the following options:
- Apply stroke on the full element (default)
- Left
- Right
- Bottom 
- Top
- Custom

**Comment by jdittrich** [comment]
* related community discussion: [Border on specific sides only](https://community.penpot.app/t/border-on-specific-sides-only/595), [
Feature Request: Stroke per side ](https://community.penpot.app/t/feature-request-stroke-per-side/2565)
* this issue [already exists on taiga](https://tree.taiga.io/project/penpot/us/2372?backlog-q=stroke&no-milestone=1) as "2372 Edit stroke properties"

**Comment by superalex** [comment]
Thank you very much!,

I'm updating Taiga to link this one too :)

**Comment by mokazemi** [comment]
Ping! Any update on it? :)

**Comment by Pterygoidien** [comment]
Hi, any update ?

**Comment by jdittrich** [comment]
pretty sure there is no update, since there is [no update on taiga](https://tree.taiga.io/project/penpot/us/2372?backlog-q=stroke&no-milestone=1) either.

**Comment by hongbo-code** [comment]
期待更新

**Comment by APattjoshi** [comment]
This is a must need hope to get an update soon, the workaround are annoying

**Comment by mrodriguer** [comment]
Can someone point to a workaround? I really need this

**Comment by garrett** [comment]
@mrodriguer: It depends on what you're doing.

It would be a lot simpler if we just had borders where the sides were specified independently, even if it would just be the width. (I usually don't even care about color being different on sides, although that could also be useful in some cases.)

But I've come up with lots of different workarounds, for different cases.

# My various workarounds

I've been adding 1px boxes aligned to edges in some cases, having them stretch on a few axis for simple groups (where a spring based layout is used), having them stretch to fill width (for horizontal ones, height for vertical ones) in flex and grid (where the row or column it's in spans across and the height/width is 1px) layouts, and have lately even used shadow without blur and with a vertical offset in some other cases.

I used to use the group with constraints a lot, until flex came around. Now with both flex and grid, I still use flex the most, but use the grid option here when laying out tables.

I hope these examples with short explanations and screenshot crops help to explain it.

Ideally, we'd just be able to set borders on objects with individual sides (or at least widths per side).

## Group with constraints

Useful for a bunch of elements without special layout. I personally don't use this much anymore, as using flex is almost always better.

![image](https://github.com/penpot/penpot/assets/10246/9750caf6-3c05-4e71-aa40-b599d442f6c7)

## Flex with auto

Useful for a bunch of objects where the size can be set by the contents, which is great for buttons or inputs.

Flex (setting "auto" stretch in different directions for each element):

![image](https://github.com/penpot/penpot/assets/10246/014f823b-a300-4844-991c-9c521b37fd85)
![image](https://github.com/penpot/penpot/assets/10246/843d41be-23f1-46a0-af70-4e0163273b79)

## Grid with span

Useful for table layouts and divider lines.

Grid (make the row span across and the rectangle "line" fill the row, then make the height of the row "auto"):

![image](https://github.com/penpot/penpot/assets/10246/d2192ae4-47e0-4e3d-974d-27b0b04306d7)
![image](https://github.com/penpot/penpot/assets/10246/4c6853c9-1042-4933-b60b-0d5979892e0d)

## Shadow (without blur)

Useful for buttons and inputs, where the size is set by the contents. This would usually be applied to the group, such as a flex, not a rectangle element.

Shadow, bottom only (change to -1 for top or change width to 1 or -1 for side):

![bottom fake border](https://github.com/penpot/penpot/assets/10246/126b64cb-e174-4e14-8c5c-8864cf9af3b6)

Shadow, using multiple (one for all around, another is above it for the bottom):

![independent fake border](https://github.com/penpot/penpot/assets/10246/d50cfd7b-5360-4b2a-ad1c-8effe721e42b)

## Hybrid: Flex using a shadow on the flex part

Use a shadow on the flex, set the text with the to-size option.

Especially good for buttons, inputs, etc. and doesn't require additional elements.

![image](https://github.com/penpot/penpot/assets/10246/8a28ff59-64e6-491a-967e-7c502ce5900d)

![image](https://github.com/penpot/penpot/assets/10246/4dd419ee-b76b-46d8-b62a-8c88e85de9a3)

**Comment by henri9813** [comment]
Hello,

No news on this on 2025 ? :)

**Comment by jhonatan-3a** [comment]
incredible how this feature has been so overloooked for years, you can beat industry standard without this

**Comment by visualcode** [comment]
I absolutely agree with @jhonatan-3a you can't claim to be the best design tool for hand-off if you don't allow a feature that is basic CSS.

**Comment by staus** [comment]
Adding my voice to this. I've just started to adopt Penpot in stead of Figma and I have been baffled that this is missing.

**Comment by joarod-eu** [comment]
Pls, this feature is a "must have" to really do the transition from Figma to Penpot

**Comment by mindplay-dk** [comment]
> Pls, this feature is a "must have" to really do the transition from Figma to Penpot

yeah, I just did a test with the exporter, and this was one of the only serious shortcomings.

as soon as this is supported, an issue should be opened in [penpot-exporter-figma-plugin](https://github.com/penpot/penpot-exporter-figma-plugin) to support it.

FYI: currently the [#1 most upvoted issue here](https://github.com/penpot/penpot/issues?q=is%3Aissue%20state%3Aopen%20sort%3Areactions-%2B1-desc) 😊

**Comment by sina-salahshour** [comment]
this issue is here from 2023 and still no one from maintainers reacted on it..

**Comment by visualcode** [comment]
I hope the Penpot team will listen to us again. This is an essential feature to have; there is not a design tool nowadays that does not include this feature to select the border-top, right, bottom, and left. It’s basic CSS, and design tokens need to assign a border colour, thickness, etc., to the border side they want. This is so essential, and as I mentioned before, you can’t claim a design tool bridges the gap between design and code if you don’t have this basic CSS property that has been in browsers for decades. Please take this feature seriously for all users.

**Comment by georges-g** [comment]
I use Penpot daily and I love the tool overall but I cannot believe this feature is still missing.

Please implement it urgently. It is pretty basic for a Web design tool and it should be a top priority.

**Comment by mindplay-dk** [comment]
yeah, I wonder why this is such a big deal 🤔

**Comment by NatachaMenjibar** [comment]
Hey, thanks for your feedback! I'm closing this feature, it will be done here #9854
